soc/riscv/riscv-privilege/common/vector.S

```diff
@@ -25,12 +25,40 @@ SECTION_FUNC(vectors, __start)
 
 	.option norvc;
 
+#if defined(CONFIG_RISCV_MTVEC_VECTORED_MODE)
+    /*
+     * Set mtvec (Machine Trap-Vector Base-Address Register)
+     * to __ivt (interrupt vector table). Add 1 to base
+     * address of __ivt to indicate that vectored mode
+     * is used (LSB = 0x1). CPU will mask the LSB out of
+     * the address so that base address of __ivt is used.
+     *
+     * NOTE: __ivt is 256-byte aligned. Incorrect alignment
+     *        of __ivt breaks this code.
+     */
+    la t0, __ivt /* Load address of interrupt vector table */
+    addi t0, t0, 1 /* Enable vectored mode by setting LSB */
+
```

Comment:
One minor nitpick: the indentation here and in the rest for your changes (4 spaces) is not consistent with the rest of this file (1 tab).

Reply:
Should be fixed.